app/models/ai_annotation.rb

```diff
@@ -62,6 +80,45 @@ def text_json=(annotation_json)
 
   private
 
+  # Extracts chunks of text from the words array, ensuring that each chunk does not exceed the specified window size.
+  def extract_chunks(words_with_newlines)
+    chunks = []
+    words_with_newlines.each_slice(WINDOW_SIZE) do |chunk_words|
+      # Concatenate words considering line breaks (determine whether to add spaces)
+      chunk_text = ""
+      chunk_words.each do |word|
+        if word == "\n"
+          chunk_text += word
+        elsif chunk_text.empty?
+          chunk_text += word
+        else
+          chunk_text += " #{word}"
+        end
+      end
+      chunks << chunk_text
+    end
+    chunks
+  end
+
+  # Extracts words from the text while preserving line breaks.
+  # Normalizes line endings (\r\n to \n) and splits each line into words.
+  # Empty lines are represented by a single newline character.
+  # Words are collected sequentially, with newline characters added as separate elements when lines end with newlines.
+  def words_with_newlines_enum(text)
+    Enumerator.new do |y|
+      normalized_text = text.gsub(/\r\n/, "\n")
+      normalized_text.each_line do |line|
+        if line.strip.empty?
+          y.yield "\n"
+        else
+          line_words = line.split(/\s+/)
+          line_words.each { |word| y.yield word }
```

Comment:
デフォルトブロック引数itをつかって
```suggestion
          line_words.each { y.yield it }
```

と書けませす。
十分に短いブロックなのでブロック引数名を省略して良いと思います。

Reply:
https://github.com/jdkim/textae_campus/pull/13/commits/339df66a87d90f68afb8c996f5f5070faecda3d6 で短縮しました。